cdap-distributions/bin/build_artifact_upload.sh

```diff
@@ -83,22 +83,35 @@ function sync_build_artifacts_to_server () {
   do
     decho "PACKAGE=${i}"
     _package=`echo ${i} | awk -F / '{ print $(NF) }'`
-    if [[ "${_package}" == *rpm ]]
-    then
-      _version_stub=`echo ${_package} | awk -F - '{ print $(NF-1) }'`
-    else
-      if [[ "${_package}" == *_* ]]
-      then
+
+    case $_package in
+
+      cdap.*_[0-9]\.[0-9]\.[0-9]\.[0-9]+.*deb | cdap.*-[0-9]\.[0-9]\.[0-9]\.[0-9]+.*rpm )  ## snapshots with timestamp
+        _version_stub=`echo ${_package} | awk -F _ '{ print $2 }'`
+        ;;
+
+      *_all.deb )                             ## debian standard
+        _version_stub=`echo ${_package} | awk -F _ '{ print $2 }' | awk -F - '{ print $1 }'`
+        ;;
+
+      *noarch.rpm )                           ## rpm standard
+        _version_stub=`echo ${_package} | awk -F - '{ print $(NF-1) }'`
+        ;;
+
+      *_*deb )                                   ## default debian case
         _version_stub=`echo ${_package} | awk -F _ '{ print $2 }'`
-      else
+        ;;
+
+      *rpm | * )                              ## default RPM case and catchall
```

Comment:
Don't assume a versioning scheme in a catchall. Make the catchall fail, since if we hit this condition, something went horribly wrong, anyway.